racket/collects/racket/match/compiler.rkt

```diff
@@ -175,16 +177,21 @@
                      (and (bound-identifier=? v v*) (or id (list v v*)))))
                  =>
                  (lambda (id)
-                   (if (identifier? id)
+                   (if (and (identifier? id) (not (inside-GSeq?)))
```

Comment:
This `(not (inside-GSeq?))` check also doesn't allow nonlinear patterns inside the "tail" of `GSeq` patterns, such as in `(list a _ ... a)` and `(cons a (list pre ... a post ...))`.

That seems too restrictive to me. In my opinion it should be restricted based on the pattern itself being under ellipses, not based on it being in the tail after something else that's under ellipses.

Reply:
I agree. This PR is too aggressive as I said at the time, but I didn't settle on a better fix.